### PR TITLE
Improve names and conventions in codegen settings

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR improves the naming and applying conventions of code generation settings. This is a preparation step towards completion of migrating off `protoc`-based code generation. Introduced changes are described below.

## `mc-java-uuid` module introduced
The new module introduces discovery of `UuidValue`s. The code generation would be added in one of the following PRs.

## Renamings
 * The proto type `CodegenSettings` was renamed to `Combined` to reflect the nature of the message and "free up" the `Settings` suffix for other names.
 * `Config` interface was renamed to `Settings`.  The implementing classes were renamed accordingly. It's not the "configuration" we deal with in this interface and classes.

## Changes to `CodegenConfig`
 * The class was renamed to `CodegenSettings`.
 * The code was converted to Kotlin.
 * Methods than were used in Java for accessing internals of the settings were transformed into properties.
 * The method `validation()` was kept and deprecated in favour of the property because it's actually used in projects applying McJava.

## Simplified user classpath for ProtoData
 * `ProtoDataConfigPlugin` now passes only McJava fat JAR artifact (`spine-mc-java-plugins`) to ProtoData user classpath. Previously, separate artifacts were added which resulted in unnecessary download of both fat JAR and separate artifacts.
 * Unused artifact values were removed from `Artifacts.kt`.

## Other notable changes
 * `mc-java-params` module was removed as redundant. Custom options are already handled by `AutoService` discovery in corresponding artifacts.
 * `PluginTestSetup` test fixture got a new method called `createProject()` so that it's possible to create parts of codegen settings without creating an instance of `CodegenSettings`.

### CODE REVIEW

File: mc-java-uuid/src/main/proto/spine/tools/mc/java/uuid/events.proto [hunk 1]
```diff
@@ -24,25 +24,27 @@
...
+
+import "spine/protodata/ast.proto";
+
+import "spine/tools/mc/java/settings.proto";
+
+// A message type which represents [UuidValue][io.spine.base.UuidValue] was discovered.
+message UuidValueDiscovered {
+
+    // The discovered type.
+    protodata.MessageType type = 2 [(required) = true];
```
Comment (armiol) [inline mc-java-uuid/src/main/proto/spine/tools/mc/java/uuid/events.proto:46]: Let's start the field indexes from `1`.

File: mc-java-uuid/src/main/proto/spine/tools/mc/java/uuid/views.proto [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+// Matches a UUID value message to corresponding code generation settings.
+message UuidMessage {
+
+    option (entity).kind = VIEW;
+
+    // The type of the message.
+    protodata.MessageType type = 1 [(required) = true];
+
+    // Code generation settings for this type.
+    spine.tools.mc.java.Uuids settings = 3 [(required) = true];
```
Comment (armiol) [inline mc-java-uuid/src/main/proto/spine/tools/mc/java/uuid/views.proto:51]: Here it should be `2`.

File: mc-java/src/test/kotlin/io/spine/tools/mc/java/settings/CodegenBlockSpec.kt [hunk 3]
```diff
@@ -132,8 +130,8 @@ class CodegenBlockSpec {
             val secondInterface = "test.iface.TestCommand"
             val fieldSuperclass = "test.cmd.Field"
             val suffix = "_my_commands.proto"
-            options.codegen { config: CodegenConfig ->
-                config.forCommands { commands: SignalConfig ->
+            options.codegen { config ->
```
Comment (armiol) [inline mc-java/src/test/kotlin/io/spine/tools/mc/java/settings/CodegenBlockSpec.kt:133]: Are they `settings` now here and below as well?

**Review by armiol — CHANGES_REQUESTED** [review]
@alexander-yevsyukov please see my comments.

File: mc-java-uuid/src/main/proto/spine/tools/mc/java/uuid/events.proto [hunk 4]
```diff
@@ -24,25 +24,27 @@
...
+option (type_url_prefix) = "type.spine.io";
+option java_package = "io.spine.tools.mc.java.uuid.event";
+option java_outer_classname = "EventsProto";
+option java_multiple_files = true;
+
+import "spine/protodata/ast.proto";
+
+import "spine/tools/mc/java/settings.proto";
+
+// A message type which represents [UuidValue][io.spine.base.UuidValue] was discovered.
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply mc-java-uuid/src/main/proto/spine/tools/mc/java/uuid/events.proto:42]: Why don't we want the verb "was" in the 1st option?

[the remark this answers is not in the document]
Comment (armiol) [reply mc-java-uuid/src/main/proto/spine/tools/mc/java/uuid/events.proto:42]: You're right. I was misreading the sentence.

**Review by armiol — APPROVED** [review]

**Pull request merged**